Mage.Sets/src/mage/cards/z/ZenosYaeGalvus.java

```diff
@@ -88,8 +88,9 @@ enum ZenosYaeGalvusPredicate implements ObjectSourcePlayerPredicate<Permanent> {
 
     @Override
     public boolean apply(ObjectSourcePlayer<Permanent> input, Game game) {
+        input.getSource().initSourceObjectZoneChangeCounter(game, false);
```

Comment:
Just saw the changes from PR. Nope, that's fully wrong. zcc sync code is part of game engine, do not use it in custom cards to "fix it". If you want to change some logic -- it's must be changed for every use, not for single card.

Reply:
We need some way to get the ZCC of the actual source object. How should it be done, then, since "getSourceObjectZoneChangeCounter" apparently does not, in fact, get the source object's current zone change counter?

Reply:
See docs above -- static and triggered abilities uses diff zcc. That's by design and that's correct. It's your use case -- card try to use same filter for both static and trigger:
![shot_250608_191139](https://github.com/user-attachments/assets/1bdcc3e5-24c7-4080-9f5d-306687199da8)

Try to use object's zcc instead ability's zcc. If you really need ability then use two diff filters/predicates.

Reply:
Find actual object's zcc: `object.getZoneChangeCounter(game)`

Reply:
I don't have the source available - the `Object` in the `ObjectSourcePlayer` is the permanent potentially being effected by the ability - but I was able to just steal the code from `AbilityImpl`'s `getCurrentSourceObjectZoneChangeCounter` and replace the calls to `getSourceObjectZoneChangeCounter` with that result.

Reply:
Related issues with source.getSourceObjectZoneChangeCounter() usage:
* #13137
* #8706
* #10479

Reply:
ability's getSourceObjectZoneChangeCounter for ability inner usage. It must be restricted to use at all from outside cause outside code don't know about ability logic.

It's started actively used from 2021. Maybe by typo))
![shot_250608_194338](https://github.com/user-attachments/assets/8d2244de-caae-4546-867b-844b176c6cc8)

Reply:
And it must be named like getStartingZoneChangeCounter

Reply:
New issue about it #13737